Add WPT for shape function parsing

smfr opened the pull request:
#### efd53559b3bf03346bb149905998bd5cba21b42f
<pre>
Add WPT for shape function parsing
<a href="https://bugs.webkit.org/show_bug.cgi?id=277346">https://bugs.webkit.org/show_bug.cgi?id=277346</a>
<a href="https://rdar.apple.com/132814350">rdar://132814350</a>

Reviewed by Tim Nguyen.

Add Basic Shape parsing and computed style tests based on <a href="https://drafts.csswg.org/css-shapes/#supported-basic-shapes.">https://drafts.csswg.org/css-shapes/#supported-basic-shapes.</a>

This adds closest-corner/farthest-corner as valid `circle()` and `ellipse()` sizes.

This also assumes that the resolution in <a href="https://github.com/w3c/csswg-drafts/issues/10645">https://github.com/w3c/csswg-drafts/issues/10645</a> passes
(which seems to be current browser behavior already).

* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/parsing/shape-outside-invalid-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/parsing/shape-outside-invalid.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/parsing/shape-outside-valid-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/parsing/shape-outside-valid.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-valid.html: Added.

Canonical link: <a href="https://commits.webkit.org/281657@main">https://commits.webkit.org/281657@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f1c34fb9e6098ed3cae5fa418d23ffa9e3c22a1a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60507 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39863 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13074 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64430 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11047 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47539 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11275 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48953 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7685 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62538 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37127 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52412 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29785 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33831 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9647 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9959 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55700 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9937 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66161 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4445 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9758 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56321 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4464 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52391 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56490 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13486 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3685 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35670 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36752 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37843 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/36497 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function.html:
```diff
@@ -0,0 +1,51 @@
+</style>
+</head>
+<body>
+<div id="target"></div>
```
Can you be consistent on whether you include the target or not?

I'd prefer not including it.

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function.html:
```diff
@@ -0,0 +1,51 @@
+    test_invalid_value("shape-outside", "circle(-10% at 20% 30%)");
+    test_invalid_value("shape-outside", "circle(1% 2% at 0% 100%)");
+
+    document.getElementById('target').remove();
```
ditto for this.

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function.html:
```diff
@@ -0,0 +1,51 @@
+    test_invalid_value("shape-outside", "circle(at 4px, 20px)");
+    test_invalid_value("shape-outside", "circle(-10px at 20px 30px)");
+    test_invalid_value("shape-outside", "circle(-10% at 20% 30%)");
+    test_invalid_value("shape-outside", "circle(1% 2% at 0% 100%)");
```
Usually we have `-valid.html` and `-invalid.html` (and also `-computed.html`) all as separate files. Can you make that consistent?

smfr committed:
Add WPT for shape function parsing
https://bugs.webkit.org/show_bug.cgi?id=277346
rdar://132814350

Reviewed by Tim Nguyen.

Add Basic Shape parsing and computed style tests based on https://drafts.csswg.org/css-shapes/#supported-basic-shapes.

This adds closest-corner/farthest-corner as valid `circle()` and `ellipse()` sizes.

This also assumes that the resolution in https://github.com/w3c/csswg-drafts/issues/10645 passes
(which seems to be current browser behavior already).

* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/parsing/shape-outside-invalid-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/parsing/shape-outside-invalid.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/parsing/shape-outside-valid-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/parsing/shape-outside-valid.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/inset-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/path-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/polygon-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/rect-function-valid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-invalid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-invalid.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-valid-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/xywh-function-valid.html: Added.

Canonical link: https://commits.webkit.org/281657@main
